### DISCUSSION TITLE
Function Parser syntax: "and" and "or" conditions in if statements; exponents

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#25161] [Q&A General] [answered]
How are these handled? I have a function in my input file that uses both and I'm getting an error. Say I wanted to type a function:

IF
(condition1 OR condition2), action1, 

ELSE
action 2

Would it be typed like:

if ((condition1 or condition 2), action1, action2) ? What is used in MOOSE to denote and/or? I tried typing them out/using && || respectively and neither worked. Specifically, I have this line of code and it's throwing an error:

`expression = 'if ( (x<((-gb_width/2)+(-z*tan(theta)))) || (x>((gb_width/2)+(z*tan(theta)))), k_bulk, k_gb)'`

Also, to raise a value to a power, is it simply the "^" character? I looked through the examples on the parsed function page and saw the expressions actually typed out (e.g x*x*x instead of x^3), so I wanted to be sure.

**Comment by GiudGiud** [comment, marked as answer]
Use | for or, and & for and

documentation:
https://fparser.readthedocs.io/en/latest/

**Reply by GiudGiud** [reply]
an example in the framework here
test/tests/meshgenerators/compute_node_position/2d.i

**Reply by richmondodufisan** [reply]
perfect, thanks!!
